### ISSUE TITLE
What is the reason of importing js npm modules in vendor.ts?

### ISSUE THREAD

**Issue by kuncevich** [preboot/angular2-webpack#76]
I just installed `d3` from npm and just import that on top of related component `.ts` file
`import * as d3 from 'd3';` I haven't put anything like `import 'd3'`; in to `vendor.ts` and without doing that I have everything working fine. 

So then I just comment all default imports in `vendor.ts`:
```
// import 'angular2/platform/browser';
// import 'angular2/platform/common_dom';
// import 'angular2/core';
// import 'angular2/router';
// import 'angular2/http';

// // RxJS
// import 'rxjs';
```
And still everything working just fine without any issues including building process.
So for me it is looks like there is no needs in `vendor.ts` in case of just javascript modules? 
Any comments on that?

FROM THE DOCS:

> How to include external css files such as bootstrap.css ?
> 
> Just install the lib and import the css files in vendor.ts. For example this is how to do it with bootstrap:
> 
> npm install bootstrap@4.0.0 --save
> And in vendor.ts add the following:
> 
> import 'bootstrap/dist/css/bootstrap.css';

**Comment by Foxandxss** [comment]
At the end, you just need one bundle output "app.js" to make your app work. The idea of having `vendor.ts` is because we split our application in three parts: `app`, `vendor` and `polyfills` (this latter one is a requirement with ng2).

Focusing on `app` and `vendor`, the idea is to have one bundle with **only** the application code and one bundle with **only** with vendor code.

To make this separation to work, we need in `vendor.ts` to load our vendors too, so when webpack sees that we did an `import 'foo'` in `app.ts` (or any .ts in our application) and in `vendor.ts`, will see that they are in common so they will get removed from the application code.

So, if you for example comment out the line where you import the router in `vendor.ts` and you import the router only in application code, it will work but the entire router code will end in the final `app.ts` bundle instead of the vendor bundle.

In short, when an import appears both in vendor.ts and application code, that 3rd party library will be removed from the application code and put in a vendor bundle alone.

This is only the first step to achieve better things like proper caching so if you update only the app code and not the vendor, you won't force the client to download the vendors again.

There were some big bugs with this latter feature in webpack so I will check if they are fixed so I can add this extra cache feature.

**Comment by kuncevich** [comment]
Cache is good. Hopefully will see it soon in `angular2-webpack`.

**Comment by RickyJiao** [comment]
@Foxandxss thanks for you comments. But if we only import some thing in `vendor.ts` but we don't use it in `app.ts`. Do we need to remove those code from `vendor.bundle.js'?

**Comment by Foxandxss** [comment]
Don't add it to the `vendor.ts` if you don't use it.
